src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -70,17 +95,37 @@ public ArduinoBoard(Stream serialPortStream)
         /// On Linux, possible values include "/dev/ttyAMA0", "/dev/serial0", "/dev/ttyUSB1", etc.</param>
         /// <param name="baudRate">Baudrate to use. It is recommended to use at least 115200 Baud.</param>
         public ArduinoBoard(string portName, int baudRate)
+        : this()
```

Comment:
would it make sense to not have a parameterless constructor at all and only have this constructor and one which takes Stream, then have an `Initialize(Stream)` ctor which is called from the Stream constructor and also in here passing the BaseStream from seriall port? I think this would clean this up a bit

Reply:
The serial port is only opened later, so the stream is not available yet. However, I can move the common stuff mostly to the Initialize function.